app/Platform/Actions/UpsertGameCoreAchievementSetFromLegacyFlags.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Actions;
+
+use App\Models\AchievementSet;
+use App\Models\Game;
+use App\Platform\Enums\AchievementSetType;
+use Illuminate\Support\Facades\DB;
+
+class UpsertGameCoreAchievementSetFromLegacyFlags
+{
+    public function execute(Game $game): void
+    {
+        $game->load(['achievements', 'gameAchievementSets']);
+
+        DB::transaction(function () use ($game) {
+            $hasCoreSets = $game->gameAchievementSets()->core()->count() > 0;
+            if ($hasCoreSets) {
+                $this->updateExistingCoreSetOfGame($game);
+            } else {
+                $newAchievementSet = $this->createNewAchievementSetFromGame($game);
+                $game->achievementSets()->attach($newAchievementSet->id, [
+                    'type' => AchievementSetType::CORE->value,
+                    'order_column' => 0,
+
+                    // Preserve the existing timestamps. Note that updated_at reuses the created_at
+                    // value, as none of the fields stored in game_achievement_sets would've changed
+                    // from the time of the first achievement promotion.
+                    'created_at' => $newAchievementSet->created_at,
+                    'updated_at' => $newAchievementSet->created_at,
+                ]);
+            }
+        });
+    }
+
+    private function createNewAchievementSetFromGame(Game $game): AchievementSet
+    {
+        list($officialAchievements, $unofficialAchievements) = $game->achievements->partition(function ($achievement) {
+            return $achievement->isPublished;
+        });
+        $allAchievements = $officialAchievements->merge($unofficialAchievements);
+
+        // First, create the set.
+        $achievementSet = AchievementSet::create([
+            'players_total' => $game->players_total,
+            'players_hardcore' => $game->players_hardcore,
+            'achievements_published' => $officialAchievements->count(),
+            'achievements_unpublished' => $unofficialAchievements->count(),
+            'points_total' => $officialAchievements->sum('points'),
+            'points_weighted' => $officialAchievements->sum('TrueRatio'),
+
+            // Preserve the existing timestamps as best as we can.
+            'created_at' => $allAchievements->min(fn ($achievement) => $achievement->DateCreated ?? now()),
+            'updated_at' => $allAchievements->max(fn ($achievement) => $achievement->DateModified ?? now()),
+        ]);
+
+        // Next, attach all the achievements to the set.
+        $allAchievements->each(function ($achievement) use ($achievementSet) {
+            $achievementSet->achievements()->attach($achievement->ID, [
+                // Preserve the existing timestamps as best as we can.
```

Comment:
should also preserve the order_column.

Reply:
Good catch, fixed in https://github.com/RetroAchievements/RAWeb/pull/2662/commits/78adfa7f91345592f30b0ab5db2c1c39a2f43303.

Reply:
Do I need to drop the table to see this change? I reran the artisan command, and I see updated timestamps on my records, but order_column is still null:
```
MariaDB [retroachievements-web]> select * from achievement_set_achievements where achievement_id=84700\G
*************************** 1. row ***************************
                id: 281293
achievement_set_id: 5760
    achievement_id: 84700
      order_column: NULL
        created_at: 2019-10-04 01:07:15
        updated_at: 2024-09-08 13:59:53
1 row in set (0.002 sec)

MariaDB [retroachievements-web]> select DisplayOrder from Achievements where ID=84700;
+--------------+
| DisplayOrder |
+--------------+
|           16 |
+--------------+
1 row in set (0.001 sec)
```